MaxText/layers/quantizations.py

```diff
@@ -104,37 +167,67 @@ def dot_general_cls(self, mesh_axes: Tuple[str, ...] = ()):
     """Returns dot_general configured with aqt params."""
     return nn.Fp8DotGeneralOp
 
+def _get_int8_quant_config(config):
+  drhs_bits = None
+  drhs_accumulator_dtype = None
+  drhs_local_aqt = None
+  if config.quantization_local_shard_count != 0:
+    drhs_bits = 8
+    drhs_accumulator_dtype = jnp.int32
+    drhs_local_aqt = aqt_config.LocalAqt(
+      contraction_axis_shard_count=config.quantization_local_shard_count
+      )
+  return aqt_config.config_v3(
+    fwd_bits=8,
+    dlhs_bits=8,
+    drhs_bits=drhs_bits,
+    rng_type="jax.uniform",
+    dlhs_local_aqt=None,
+    drhs_local_aqt=drhs_local_aqt,
+    fwd_accumulator_dtype=jnp.int32,
+    dlhs_accumulator_dtype=jnp.int32,
+    drhs_accumulator_dtype=drhs_accumulator_dtype,
+    )
+
+
+def _get_weight_only_quant_config(lhs_bits=None, rhs_bits=None):
+  return aqt_config.dot_general_make(lhs_bits=lhs_bits, rhs_bits=rhs_bits)
+
+
+def _get_mixed_precision_quant_config(config, config_file):
+  """Set quantization params based on user configuration."""
+  with open(config_file, "r") as infile:
+    mixed_precision_config = json.load(infile)
+  ret_config = {}
+  ret_config["default"] = [aqt_config.dot_general_make(lhs_bits=None, rhs_bits=8), -1]
+  for layer_name_re, layer_quantization_config in mixed_precision_config.items():
+    rhs_num_bits = layer_quantization_config.get("bits", 8)
+    tile_size = layer_quantization_config.get("tile_size", -1)
```

Comment:
can we include documentation in the base.yml of the defaults and the behavior of the defaults - that tile_size=-1 signifies "don't tile"

Reply:
Done